# Type: test
# Subtype: T1
# English Summary: Create a parent folder as a repository to store files
# Chinese Summary: 创建父文件夹作为存储文件的存储库
# Entry Number: 14

# === Git Diff Content ===

diff --git a/camel-core/src/test/java/org/apache/camel/processor/FileIdempotentConsumerCreateRepoTest.java b/camel-core/src/test/java/org/apache/camel/processor/FileIdempotentConsumerCreateRepoTest.java
index bdfd923551..caa44a33cc 100644
--- a/camel-core/src/test/java/org/apache/camel/processor/FileIdempotentConsumerCreateRepoTest.java
+++ b/camel-core/src/test/java/org/apache/camel/processor/FileIdempotentConsumerCreateRepoTest.java
@@ -19,6 +19,8 @@ package org.apache.camel.processor;
 import java.io.File;
 import java.io.IOException;
 
+import static java.util.UUID.randomUUID;
+
 import org.apache.camel.spi.IdempotentRepository;
 import org.junit.Assert;
 import org.junit.Test;
@@ -30,9 +32,8 @@ public class FileIdempotentConsumerCreateRepoTest extends Assert {
     @Test
     public void shouldCreateParentOfRepositoryFileStore() throws IOException {
         // Given
-        File tmpDir = new File(System.getProperty("java.io.tmpdir"));
-        File storeParent = new File(tmpDir, "repositoryParent");
-        File store = new File(storeParent, "repository");
+        File parentDirectory = new File("target/repositoryParent_" + randomUUID());
+        File store = new File(parentDirectory, "store");
         IdempotentRepository<String> repo = fileIdempotentRepository(store);
 
         // When
